prediction_market_agent/agents/blockchain_coding_agent/functions.py

```diff
@@ -75,62 +75,45 @@ def execute_read_function(
         contract_address (str): The address of the smart contract on which to execute the function.
         abi (str): The ABI (Application Binary Interface) of the smart contract.
         function_name (str): The name of the function to execute on the smart contract.
-        function_parameters (list): A list of parameters to pass to the function.
+        function_params (list): A list of parameters to pass to the function.
 
     Returns:
         Any: The result of calling the specified function on the smart contract.
 
     """
-
-    c = ContractOnGnosisChain(
-        abi=abi_field_validator(abi), address=Web3.to_checksum_address(contract_address)
-    )
-    return c.call(function_name, function_parameters)
-
-
-def register_all_functions(
-    caller_agent: ConversableAgent, executor_agent: ConversableAgent
-) -> None:
-    # Register the search tool.
-    register_function(
-        tavily_search,
-        caller=caller_agent,
-        executor=executor_agent,
-        name="search_tool",
-        description="Search the web for the given query",
+    rpc_endpoint = get_rpc_endpoint()
+    w3 = Web3(Web3.HTTPProvider(rpc_endpoint))
+    contract = w3.eth.contract(
+        address=Web3.to_checksum_address(contract_address), abi=abi_field_validator(abi)
     )
 
-    # Register the calculator function to the two agents.
-    register_function(
-        fetch_read_methods_from_blockscout,
-        caller=caller_agent,  # The assistant agent can suggest calls to the calculator.
-        executor=executor_agent,  # The user proxy agent can execute the calculator calls.
-        name="fetch_read_methods_from_blockscout",  # By default, the function name is used as the tool name.
-        description="Function for fetching the ABI from a verified smart contract on Gnosis Chain",
-        # A description of the tool.
-    )
+    output = contract.functions[function_name](
+        *parse_function_params(function_params)
+    ).call()  # type: ignore # TODO: Fix Mypy, as this works just OK.
+    return output
```

Comment:
Why didn't you use `call_function_on_contract` inside of this function instead of copy-pasting the code from PMAT?

Reply:
My rationale was not to use PMAT to allow the agent to rely purely on web3py (I was considering the case where the agent also writes some code).
However, here it's purely doing function calling, so sure, let's use PMAT for simplified logic.
Implement this also for the other comment.